core/src/main/java/smashdudes/graphics/effects/ParticleEmitter.java

```diff
@@ -120,18 +113,23 @@ private static void updateParticle(ParticleEmitterConfig config, Particle p, flo
         p.life -= dt;
     }
 
-    private static void drawParticle(SpriteBatch sb, Particle p)
+    public boolean depleted()
     {
-        float t = 1.0f  - p.life / p.initialLife;
+        return elapsedTime > config.emissionDuration && activeParticles.size == 0;
+    }
```

Comment:
Does this allow for infinite spawning? 
There should be some value for `config.emissionDuration` that makes it so depleted always returns false. 
For instance, a projectile may exist until it is depleted in code, so we may not know the exact lifetime of it.

Reply:
No it will not infinitely spawn, depleted is not used to determine when the emitter is finished creating particles, but to determine when all of the particles it is, and will be. responsible for are gone (since it handles updating the particles until they are dead). The elapsed time check is just to make sure you aren't determining this too early, since it's possible that more will spawn.

The idea was to check if the emitter has been fully depleted (i.e., there are no more particles associated with this emitter).

We can set a final static constant in the file so sure.

Reply:
What I meant was we _should_ allow infinite spawning to occur of the emitter is configured to do so.